File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        // we can go binary-search/two-pointers/
        int left = 0, right = numbers.length - 1;
        while(left != right) {
            int sum = numbers[left] + numbers[right];
            if (sum == target) {
                return new int[]{left + 1, right + 1};
            }
            else if (sum > target) {
                right--;
            }
            else
                left++;
        }
        return null;
    }
}

File: Data Structures & Algorithms/find-target-in-rotated-sorted-array/submission-1.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int left = 0;
        int right = nums.size() - 1;
        while(left <= right)
        {
            int mid = (left + right) / 2;
            if (nums[mid] == target)
                return mid;
            if (nums[left] <= nums[mid])//left side
            {
                if (target < nums[left] || target > nums[mid])//look right
                {
                    left = mid + 1;
                }
                else//look left
                {
                    right = mid - 1;
                }
            }   
            else //right side
            {
                if (target > nums[right] || target < nums[mid])//look left
                {
                    right = mid - 1;
                }
                else//look right
                {
                    left = mid + 1;
                }
            }        
       }
       return -1;
    }
};